x-pack/plugins/security_solution/public/explore/containers/risk_score/kpi/index.tsx

```diff
@@ -73,18 +80,42 @@ export const useRiskScoreKpi = ({
   const isModuleDisabled = !!error && isIndexNotFoundError(error);
 
   useEffect(() => {
-    if (!skip && defaultIndex && featureEnabled) {
+    if (
+      !skip &&
+      defaultIndex &&
+      !isDeprecatedLoading &&
+      isAuthorized &&
+      isEnabled &&
+      !isDeprecated
```

Comment:
Apply the same check we do when fetching risk scores: https://github.com/elastic/kibana/blob/1ba327a93ad5e751360e5ec66b0df75cc073e5ac/x-pack/plugins/security_solution/public/explore/containers/risk_score/all/index.tsx#L198-L204